org-ros.el

```diff
@@ -43,24 +43,47 @@
 (defcustom org-ros-secondary-screencapture-switch "-s"
   "Secondary screencapture software selection switch."
   :type 'string)
- 
+
+(defcustom org-ros-windows-screencapture "powershell.exe"
+  "Powershell screencapture software (for people using wsl2)."
+  :type 'string)
+
+(defcustom org-ros-subdirectory ""
+  "Subdirectory to store screenshots."
+  :type 'string)
+
+(defconst org-ros-dir (file-name-directory (or load-file-name buffer-file-name)))
+
+
 ;;;###autoload
 (defun org-ros ()
   "Screenshots an image to an org-file."
   (interactive)
   (if buffer-file-name
       (progn
         (message "Waiting for region selection with mouse...")
-        (let ((filename
-               (concat "./"
-                       (file-name-nondirectory buffer-file-name)
-                       "_"
-                       (format-time-string "%Y%m%d_%H%M%S")
-                       ".png")))
-          (if (executable-find org-ros-primary-screencapture)
-	      (call-process org-ros-primary-screencapture nil nil nil org-ros-primary-screencapture-switch filename)
-	    (call-process org-ros-secondary-screencapture nil nil nil org-ros-secondary-screencapture-switch filename))
-          (insert "[[" filename "]]")
+        (if org-ros-subdirectory
+            (make-directory org-ros-subdirectory t))
+        (let* ((default-filename (concat
+                                  (file-name-nondirectory buffer-file-name)
+                                  "_"
+                                  (format-time-string "%Y%m%d_%H%M%S")
+                                  ".png"))
+               (display-name (read-string
+                              (format "Name the screenshot (default: \"%s\"): " default-filename)
+                              nil nil default-filename))
+               (filepath (file-name-concat (file-name-directory buffer-file-name)
```

Comment:
Nice, I just tested and this way of making a dir (via elisp) is also working on macOS.